subgraphs/makerdao/src/common/helpers.ts

```diff
@@ -438,10 +437,17 @@ export function updatePosition(
     }
 
     lenderPosition.balance = lenderPosition.balance.plus(deltaCollateral);
-    assert(
-      lenderPosition.balance.ge(BIGINT_ZERO),
-      `[updatePosition]balance for position ${lenderPosition.id} ${lenderPosition.balance} < 0`,
-    );
+    if (lenderPosition.balance.le(BIGINT_ZERO)) {
+      if (lenderPosition.balance.ge(BIGINT_NEG_HUNDRED)) {
+        // a small negative lender position, likely due to rounding
+        lenderPosition.balance = BIGINT_ZERO;
+      } else {
+        log.error("[updatePosition]balance for position {} = {} < 0", [
```

Comment:
should anything be done if pos balances are "very" negative?

Reply:
I cannot think of a case where this would happen and is reasonable/correct data, which is why I log an error here. I can of course do `log.critical()` to stop the subgraph here. Do you prefer the latter?

Reply:
tbh I really prefer to not add/subtract position balances and TVL stuff bc it doesn't account for interest. But if there are no contract calls that would give us the position balance of an account, this is okay.

Reply:
That's a good point. I can switch to get position balance using contract call.

Reply:
This doesn't work. It seems there are cases of multiple transactions in the same block & getting position balance from contract call gets the final balance. I am reverting. I think we should fix the this current issue & come back to do the refectory later, e.g. when upgrading schema to 3.0. What do you think, @dmelotik ?

BTW, I plan to re-do the prettier formating (I used an old config) once you're done reviewing, because I don't want to show thousands of formatting changes.

Reply:
Okay, let's leave as is @tnkrxyz. Feel free to do the prettier update in a separate commit. I will approve and you can merge when the checks pass